backend/src/crd/controller/scheduledworkflow/controller.go

```diff
@@ -457,65 +465,138 @@ func (c *Controller) submitNextWorkflowIfNeeded(ctx context.Context, swf *util.S
-		return nil, nextScheduledEpoch, err
+		return false, nextScheduledEpoch, err
 	}
 	log.WithFields(log.Fields{
 		ScheduledWorkflow: swf.Name,
-		Workflow:          workflow.Get().Name,
+		Workflow:          workflowName,
 	}).Infof("Submitting workflow for ScheduledWorkflow (%v): workflow (%v) successfully submitted (scheduled at: %v)",
-		swf.Name, workflow.Get().Name, commonutil.FormatTimeForLogging(nextScheduledEpoch))
-	return workflow, nextScheduledEpoch, nil
+		swf.Name, workflowName, commonutil.FormatTimeForLogging(nextScheduledEpoch))
+	return submitted, nextScheduledEpoch, nil
 }
 
 func (c *Controller) submitNewWorkflowIfNotAlreadySubmitted(
 	ctx context.Context,
 	swf *util.ScheduledWorkflow, nextScheduledEpoch int64, nowEpoch int64) (
-	*commonutil.Workflow, error) {
+	bool, string, error) {
 
 	workflowName := swf.NextResourceName()
 
 	// Try to fetch this workflow
 	// If it already exists, it means that it was already created in a previous iteration
 	// of this controller but that the controller failed to save this data.
-	foundWorkflow, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
+	_, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
 		workflowName)
 	if err == nil {
 		// The workflow was already created by a previous iteration of this controller.
 		// Nothing to do except returning the information needed by the controller to update
 		// the ScheduledWorkflow status.
-		return foundWorkflow, nil
+		return true, workflowName, nil
 	}
 
 	if !isNotFoundError {
 		// There was an error while attempting to retrieve the workflow
-		return nil, err
+		return false, "", err
 	}
 
-	// If the workflow is not found, we need to create it.
-	newWorkflow, err := swf.NewWorkflow(nextScheduledEpoch, nowEpoch)
-	createdWorkflow, err := c.workflowClient.Create(ctx, swf.Namespace, newWorkflow)
+	labels := map[string]string{
+		commonutil.LabelKeyWorkflowScheduledWorkflowName:      swf.Name,
+		commonutil.LabelKeyWorkflowEpoch:                      commonutil.FormatInt64ForLabel(nextScheduledEpoch),
+		commonutil.LabelKeyWorkflowIndex:                      commonutil.FormatInt64ForLabel(swf.NextIndex()),
+		commonutil.LabelKeyWorkflowIsOwnedByScheduledWorkflow: "true",
+	}
 
-	if err != nil {
-		return nil, err
+	// Needed for backward compatibility, this should be removed in the future
+	if swf.Spec.PipelineVersionID == nil && swf.Spec.PipelineID == nil {
+		newWorkflow, err := swf.NewWorkflow(nextScheduledEpoch, nowEpoch)
+		if err != nil {
+			return false, "", err
+		}
+
+		_, err = c.workflowClient.Create(ctx, swf.Namespace, newWorkflow)
+		return true, workflowName, err
+
+	}
+
+	formatter := commonutil.NewSWFParameterFormatter("", nextScheduledEpoch, nowEpoch, swf.NextIndex())
+	formattedParams := formatter.FormatWorkflowParameters(swf.GetWorkflowParametersAsMap())
+	parameters := OverrideParameters(swf.Spec.Workflow.Spec.Arguments.Parameters, formattedParams)
```

Comment:
Note, parameters will be a different format now for v2.
https://github.com/kubeflow/pipelines/blob/60233f2a1da9e206ab0077193f091a8d39dcc05a/backend/api/pipeline_spec.proto#L47

That's why I mentioned adding a field in swf CRD that exactly mimics the PipelineSpec object.

Curious about your thoughts

Reply:
See your point and agree that we should have it as a seperate field with a map[string]string.